extensions-contrib/ambari-metrics-emitter/src/main/java/io/druid/emitter/ambari/metrics/AmbariMetricsEmitter.java

```diff
@@ -0,0 +1,234 @@
+        }
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            config.getFlushPeriod(),
+            config.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+
+  @Override
+  public void emit(Event event)
+  {
+    if (!started.get()) {
+      throw new ISE("WTF emit was called while service is not started yet");
+    }
+    if (event instanceof ServiceMetricEvent) {
+      final TimelineMetric timelineEvent = timelineMetricConverter.druidEventToTimelineMetric((ServiceMetricEvent) event);
+      if (timelineEvent == null) {
+        return;
+      }
+      try {
+        final boolean isSuccessful = eventsQueue.offer(
+            timelineEvent,
+            config.getEmitWaitTime(),
+            TimeUnit.MILLISECONDS
+        );
+        if (!isSuccessful) {
+          if (countLostEvents.getAndIncrement() % 1000 == 0) {
+            log.error(
+                "Lost total of [%s] events because of emitter queue is full. Please increase the capacity or/and the consumer frequency",
+                countLostEvents.get()
+            );
+          }
+        }
+      }
+      catch (InterruptedException e) {
+        log.error(e, "got interrupted with message [%s]", e.getMessage());
+        Thread.currentThread().interrupt();
+      }
+    } else if (event instanceof AlertEvent) {
+      for (Emitter emitter : emitterList) {
+        emitter.emit(event);
+      }
+    } else {
+      throw new ISE("unknown event type [%s]", event.getClass());
+    }
+  }
+
+  @Override
+  protected String getCollectorUri()
+  {
+    return collectorURI;
+  }
+
+  @Override
+  protected int getTimeoutSeconds()
+  {
+    return (int) (FLUSH_TIMEOUT / 1000);
+  }
+
+  private class ConsumerRunnable implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      try {
+        int batchSize = config.getBatchSize();
+        TimelineMetrics metrics = new TimelineMetrics();
+        while (eventsQueue.size() > 0 && !exec.isShutdown()) {
+          try {
+            final TimelineMetric metricEvent = eventsQueue.poll(
+                config.getWaitForEventTime(),
+                TimeUnit.MILLISECONDS
+            );
+            if (metricEvent != null) {
+              metrics.addOrMergeTimelineMetric(metricEvent);
+              if (metrics.getMetrics().size() == batchSize) {
+                emitMetrics(metrics);
+                log.debug(
+                    "sent [%d] events",
+                    metrics.getMetrics().size()
+                );
+                metrics = new TimelineMetrics();
+              }
+            }
+          }
+          catch (InterruptedException e) {
+            log.error(e, e.getMessage());
+            if (e instanceof InterruptedException) {
+              Thread.currentThread().interrupt();
+            }
+          }
+          if (metrics.getMetrics().size() > 0) {
```

Comment:
isn't this going to report each event individually and not wait for the batch to complete?